internal/parser/planparserv2/convert_field_data_to_generic_value.go

```diff
@@ -0,0 +1,126 @@
+package planparserv2
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/internal/proto/planpb"
+)
+
+func ConvertFieldDataToGenericValue(data *schemapb.ScalarField, dataType schemapb.DataType) ([]*planpb.GenericValue, error) {
+	if data == nil {
+		return nil, fmt.Errorf("convert field data is nil")
```

Comment:
use merr `expression placeholder value is nil, placeholder name: {name}`

Reply:
got it